src/Sulu/Component/Content/Types/Link.php

```diff
@@ -25,8 +28,10 @@ class Link extends SimpleContentType
 {
     public const LINK_TYPE_EXTERNAL = 'external';
 
-    public function __construct(private LinkProviderPoolInterface $providerPool)
-    {
+    public function __construct(
+        private LinkProviderPoolInterface $providerPool,
+        private ReferenceStorePoolInterface $referenceStorePool,
```

Comment:
the reference pool need to be optional else it would be a bc break.

```suggestion
        private ?ReferenceStorePoolInterface $referenceStorePool = null,
```

we should throw a deprecation message in the constructor.


```php
        if (!$this->referenceStorePool instanceof ReferenceStorePoolInterface) {
            @trigger_deprecation('sulu/sulu', '2.5', 'Initializing "' . __CLASS__ . '" without referenceStorePool is deprecated.');
        }
```